### PR TITLE
[Story] Make courses editable, deletable

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: frontend/components/elements/CourseItem.vue [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    View projects, topics and sprints
+                </router-link>
+            </template>
+
+            <CourseForm
+                v-else
+                no-semester
+                v-model="editingData"
+                @cancel="editing = false"
+                @submit="saveCourse"/>
```
Comment (anjakammer) [inline frontend/components/elements/CourseItem.vue:52]: I cannot save a course, after editing. Does not matter whether I am editing a field or not. There are no error messages.

File: frontend/components/elements/CourseItem.vue [hunk 2]
```diff
@@ -0,0 +1,148 @@
+<template>
+    <li class="course-item card">
+        <div class="card-body">
+            <template v-if="!editing">
+                <BButton
+                    size="sm"
+                    variant="outline-danger"
+                    class="float-right ml-1"
+                    v-confirm="{ action: deleteCourse, text: 'Are you sure you want to delete this course?' }">
```
Comment (anjakammer) [inline frontend/components/elements/CourseItem.vue:9]: This confirmation comes up 2 times, but is deleted at the first request.

**Review by anjakammer — CHANGES_REQUESTED** [review]
Some tiny things regarding validation and deletion confirmation. :see_no_evil:

File: frontend/components/elements/CourseItem.vue (hunk 2 above)
Comment (ghost) [reply frontend/components/elements/CourseItem.vue:9]: there's a ticket for that. it's the same bug for every resource, no need to record it for all of them individually..

File: frontend/components/elements/CourseItem.vue (hunk 1 above)
Comment (ghost) [reply frontend/components/elements/CourseItem.vue:52]: @anjakammer ah shoot, I forgot to adjust the semester validation again after fixing that. should work now.

**Review by anjakammer — APPROVED** [review]

**Pull request merged**